bin/describe.ml

```diff
@@ -0,0 +1,203 @@
+open Stdune
+open Import
+
+let doc = "Describe the workspace."
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P
+      {|Describe what is in the current workspace in either human or
+        machine readable form.
+
+        By default, this command output a human readable description of
+        the current workspace. This output is aimed at human and is not
+        suitable for machine processing. In particular, it is not versioned.
+
+        If you want to interpret the output of this command from a program,
+        you must use the $(b,--format) option to specify a machine readable
+        format as well as the $(b,--lang) option to get a stable output.|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Term.info "describe" ~doc ~man
+
+(* Crawl the workspace to get all the data *)
+module Crawl = struct
+  open Dune
+
+  let uid_of_library lib =
+    Digest.generic (Lib.name lib, Lib_info.src_dir (Lib.info lib))
```

Comment:
To be extra careful, maybe it's best to convert paths to strings here.

Reply:
Sounds good, done